src/EnergyPlus/MicroturbineElectricGenerator.cc

```diff
@@ -996,193 +842,137 @@ namespace MicroturbineElectricGenerator {
-
-                SetupOutputVariable("Generator Ancillary Electric Energy",
-                                    OutputProcessor::Unit::J,
-                                    MTGeneratorReport(GeneratorNum).AncillaryEnergy,
-                                    "System",
-                                    "Sum",
-                                    MTGenerator(GeneratorNum).Name);
-            }
-            //   Report combustion air outlet conditions if exhaust air calculations are active
-            if (MTGenerator(GeneratorNum).ExhAirCalcsActive) {
-                SetupOutputVariable("Generator Exhaust Air Mass Flow Rate",
-                                    OutputProcessor::Unit::kg_s,
-                                    MTGeneratorReport(GeneratorNum).ExhAirMassFlowRate,
-                                    "System",
-                                    "Average",
-                                    MTGenerator(GeneratorNum).Name);
-                SetupOutputVariable("Generator Exhaust Air Temperature",
-                                    OutputProcessor::Unit::C,
-                                    MTGeneratorReport(GeneratorNum).ExhAirTemperature,
-                                    "System",
-                                    "Average",
-                                    MTGenerator(GeneratorNum).Name);
-            }
+            SetupOutputVariable(
+                "Generator Exhaust Air Temperature", OutputProcessor::Unit::C, this->ExhaustAirTemperature, "System", "Average", this->Name);
         }
     }
 
-    // End of Get Input subroutine for the MT Generator Module
-    //******************************************************************************
+    void MTGeneratorSpecs::simulate(const PlantLocation &EP_UNUSED(calledFromLocation),
+                                    bool EP_UNUSED(FirstHVACIteration),
+                                    Real64 &EP_UNUSED(CurLoad),
+                                    bool EP_UNUSED(RunFlag))
+    {
+        // empty function to emulate current behavior as of conversion to using the PlantComponent calling structure.
+        // calls from the plant side... do nothing.
+        // calls from the ElectricPowerServiceManger call the init, calc, and update worker functions
+    }
 
-    // Begin MT Generator Module Initialize Subroutine
-    // *****************************************************************************
+    void MTGeneratorSpecs::getDesignCapacities(const PlantLocation &EP_UNUSED(calledFromLocation), Real64 &MaxLoad, Real64 &MinLoad, Real64 &OptLoad)
+    {
+        MaxLoad = 0.0;
+        MinLoad = 0.0;
+        OptLoad = 0.0;
+    }
 
-    void InitMTGenerators(int const GenNum,
-                          bool const RunFlag,
-                          Real64 const MyLoad, // electrical load in W
-                          bool const FirstHVACIteration)
+    void MTGeneratorSpecs::InitMTGenerators(bool const RunFlag,
```

Comment:
Just an editorial note, feel free to rename these functions to simple names like `initialize()` and `calculate()` as you wrap up the final touches.  Not necessary for this one.